config_utilities_ros/demos/demo_ros_dynamic_config_server.cpp

```diff
@@ -0,0 +1,242 @@
+}
+
+// Declare an object with a dynamic config.
+class ObjectWithDynamicConfig {
+ public:
+  explicit ObjectWithDynamicConfig(const std::string& name, const MyConfig& initial_config = MyConfig())
+      : name_(name), config_(name, initial_config) {
+    // The above initialization registers the dynamic config with its global identifier nam, where global is within the
+    // process and will be resolved to unique name for every node/object. The config is initialized with the initial
+    // parameters. The callback is called whenever the config is updated.
+    config_.setCallback(std::bind(&ObjectWithDynamicConfig::callback, this));
+  }
+
+ private:
+  const std::string name_;
+  config::DynamicConfig<MyConfig> config_;
+  std::vector<std::unique_ptr<BaseModule>> modules_;
+
+  void createAndPrintModules(const MyConfig& config) {
+    // Create all modules from the config.
+    modules_.clear();
+    if (config.virtual_module) {
+      modules_.emplace_back(config.virtual_module.create());
+    }
+    for (const auto& module : config.modules) {
+      modules_.emplace_back(module.create());
+    }
+    for (const auto& [name, module] : config.module_map) {
+      modules_.emplace_back(module.create());
+    }
+
+    // Have all modules print.
+    for (const auto& module : modules_) {
+      module->print();
+    }
+  }
+
+  void callback() {
+    // Do something with the new config.
+    // For thread safety, dynamic configs always need to be accessed via the get() method., ideally once per callback.
+    const auto config = config_.get();
+    std::cout << "Received new config for " << name_ << ":\n" << config::toString(config) << std::endl;
+    createAndPrintModules(config);
+  }
+};
+
+class DemoNode : public rclcpp::Node {
+ public:
+  DemoNode() : Node("demo_node"), obj1_("object_in_the_node"), server_(this) {}
+
+ private:
+  // Dynamic config objects can also be declared in the node.
+  ObjectWithDynamicConfig obj1_;
+
+  // Anywhere in the node or outside, a dynamic config server can be created which will manage all dynamic configs
+  // throughout the node. In the node is preferred, as lifetime will be managed automatically.
+  const config::RosDynamicConfigServer server_;
+};
+
+}  // namespace demo
+
+int main(int argc, char** argv) {
+  rclcpp::init(argc, argv);
+  google::InstallFailureSignalHandler();
+
+  // Create some objects with a dynamic config.
+  demo::ObjectWithDynamicConfig obj("dynamic_config_object");
+  demo::ObjectWithDynamicConfig other_obj("other_object");
+
+  // All standalone dynamic configs will equally be registered.
+  config::DynamicConfig<demo::SubConfig> config("standalone_config");
+
+  // Create a ROS node. Dynamic configs as well as the server can also directly live in the node.
+  auto node = std::make_shared<demo::DemoNode>();
+
+  // Alternative to the server living in the node, it could also be created here. Note that only one server should be
+  // created per node.
+  if (false) {
```

Comment:
I'd leave this commented out instead of using the always false if statement (which sometimes can trigger compiler warnings)